### ISSUE TITLE
Explainability - Phase 1

### ISSUE THREAD

**Issue by andrewballantyne** [opendatahub-io/odh-dashboard#1127]
* UX Effort: https://github.com/opendatahub-io/sig-ml-developer-experience/issues/29

**Target Branch**
f/mserving-metrics

## Foundation Work
* #1001 
* #1151
* #1152 

## P0: Ability to navigate to a tab dedicated to model bias reporting for each served modelConfigured bias reports are unique to a served model.
* #1001 

## P0: Ability to opt in to and configure bias detection for models already in production

> Display message indicating model bias reporting is not configured 
> Provide option to configure bias metric reporting
> Give users list of available fairness metrics
>User to select a metric to monitor over the lifetime of the versioned model
>Configure selected fairness metric, calculation and bias detection
> User inputs the sensitive feature/group/label/column/facet and corresponding values applicable to their model
>Default to Trusty-provided mathematically sensible baseline defaults and thresholds 
>Give user the ability to update thresholds to their preferred values
>Configure metric resolution 
>User provided name for monitoring job
>Allow user to schedule metric calculation job cadence
>User has ability to update default interval 
>Button to save or cancel configuration
>Store configuration state for use in dashboard
>Configured dashboards persist over the lifetime of the served model version 
>Return to main page, ability to add new configuration
>Allow user to configure multiple facets for any of their selected metrics For example, F1 and F2 for SPD, but only F1 for DIR

* #1001
* #1153
* #1161
* #1269
* #1275
* #1267
* #1154
* #1274

## P0: Configured bias detection graphs over time
>Clearly state the bias metric in the graph title
>User can select which group/label/column to view from dropdown
>User friendly description of monitoring job
>Display calculated metrics over dates and time 
>Render baseline and threshold values and clearly indicate threshold violations
>Ability to update the graph’s refresh rate
>Adjust viewable time range, for example: 30 minutes to 3 days
* #1001 
* #1273
* #1163
* #1158
* #1159
* #1160 

## P0: Ability to quickly reference bias detection documentation within product
>Hover for description of metric and it’s usage
* #1160 
* Possibly requires one more ticket when mocks available.

## P0: Ability to update report as needed
>User can update, add or remove metric configurations
>User can add a metric configurations, same as above
>User can remove/delete a configuration. Configuration instance will no longer be calculated nor displayed
>User can update variables used in configuration instance. E.G.
>User can update calculation resolution - 5000 -> 1000 observations
>User can change threshold
>Updates will recalculate and display new report
>Current configuration persisted and used going forward
* #1153
* #1161
* #1162
* #1163
* #1274 

## Additional Requirement [19/05/2023]: 
> User can add TrustyAI service per Data Science Project
> User can edit TrustyAI service details
> User can delete TrustyAI service per Data Science Project
* #1276

## Supplementary work
* #1336
* #1359
* #1383 
* #1384
* #1386
* #1403
* #1472

## Additional issues raised
* #1447
* #1511
* #1531
* #1532
* #1533
* #1996
* #2022
* #2023
* #2024
* #2088
* #2253

After incubation begins:
* #1526
* #2087

## Dependent on:
* #1854

**Comment by alexcreasy** [comment]
Update: a PR has been raised for UWM (https://github.com/opendatahub-io/odh-dashboard/issues/1854) it is awaiting review and unblocks development of any remaining tickets for Explainability.

**Comment by alexcreasy** [comment]
Update: development is in progress on the remaining issues.

**Comment by alexcreasy** [comment]
Update: Still working through the last couple of issues, we have one new ticket before incubation, which is #2088

**Comment by dgutride** [comment]
All subtasks migrated - moving tracker to closed
